cmd/uhc/account/users/cmd.go

```diff
@@ -0,0 +1,238 @@
+
+	"github.com/openshift-online/uhc-cli/pkg/config"
+	"github.com/openshift-online/uhc-cli/pkg/util"
+	"github.com/openshift-online/uhc-sdk-go/pkg/client"
+	amv1 "github.com/openshift-online/uhc-sdk-go/pkg/client/accountsmgmt/v1"
+
+	"github.com/spf13/cobra"
+)
+
+var args struct {
+	debug bool
+	org   string
+}
+
+var Cmd = &cobra.Command{
+	Use:   "users",
+	Short: "Retrieve users and their roles",
+	Long:  "Retrieve information of all users/roles in the same organization",
+	Run:   run,
+}
+
+func init() {
+	// Add flags to rootCmd:
+	flags := Cmd.Flags()
+	flags.BoolVar(
+		&args.debug,
+		"debug",
+		false,
+		"Enable debug mode.",
+	)
+	flags.StringVar(
+		&args.org,
+		"org",
+		"", // Default value gets assigned later as connection is needed.
+		"Organization ID, deafaults to current users organization.",
+	)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+
+	// Load the configuration file:
+	cfg, err := config.Load()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't load config file: %v\n", err)
+		os.Exit(1)
+	}
+	if cfg == nil {
+		fmt.Fprintf(os.Stderr, "Not logged in, run the 'login' command\n")
+		os.Exit(1)
+	}
+
+	// Check that the configuration has credentials or tokens that haven't have expired:
+	armed, err := config.Armed(cfg)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't check if tokens have expired: %v\n", err)
+		os.Exit(1)
+	}
+	if !armed {
+		fmt.Fprintf(os.Stderr, "Tokens have expired, run the 'login' command\n")
+		os.Exit(1)
+	}
+
+	// Create the connection:
+	logger, err := util.NewLogger(args.debug)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't create logger: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Create the connection, and remember to close it:
+	connection, err := client.NewConnectionBuilder().
+		Logger(logger).
+		TokenURL(cfg.TokenURL).
+		Client(cfg.ClientID, cfg.ClientSecret).
+		Scopes(cfg.Scopes...).
+		URL(cfg.URL).
+		User(cfg.User, cfg.Password).
+		Tokens(cfg.AccessToken, cfg.RefreshToken).
+		Insecure(cfg.Insecure).
+		Build()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't create connection: %v\n", err)
+		os.Exit(1)
+	}
+	defer connection.Close()
+
+	// needed variables:
+	pageSize := 100
+	pageIndex := 1
+	namePad := 40
+
+	// Organization to search in case one was not provided:
+	if args.org == "" {
+		// Get organization of current user:
+		userConn, err := connection.AccountsMgmt().V1().CurrentAccount().Get().
+			Send()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Can't retrieve current user information: %v\n", err)
+			os.Exit(1)
+		}
+		userOrg, _ := userConn.Body().GetOrganization()
+		args.org = userOrg.ID()
+	}
+
+	// Print top.
+	fmt.Println(stringPad("USER", namePad), "ROLES")
+	fmt.Println()
+
+	// Display a list of all users in our organization and their roles:
+	for {
+		// Get list of accounts in organization.
+		usersConn := connection.AccountsMgmt().V1().Accounts().List().Size(pageSize).Page(pageIndex)
+
+		// Format search request:
+		searchRequest := ""
```

Comment:
This initialization is redundant, as you are immediately assigning another value.

Also the `...Request` suffix it as bit confusing, as there are other requests in this context. I'd say that this is rather a `searchQuery`.

I'd suggest that you first calculate this query, and then send the request without creating the intermediate request variable:

```go
// Format search query:
searchQuery := ...

// Get all the users within the organization:
usersResponse, err := connection.AccountsMgmt().V1().Accounts().List().
        Size(pageSize).
        Page(pageIndex).
        Search(searchQuery).
        Send()
if err != nil {
        ...
}
```